docs/maps/reverse-geocoding-tutorial.asciidoc

```diff
@@ -0,0 +1,175 @@
+
+[float]
+=== Step 3: Reverse geocoding
+To visualize CSA regions by web log traffic, the web log traffic must contain a CSA region identifier. We will use {es} {ref}/enrich-processor.html[enrich processor] to add CSA region identifiers to Kibana's web logs sample data set. You can skip this step if your source data already contains region identifies.
+
+. Open the main menu, then click *Dev Tools*.
+. In *Console*, create a {ref}/geo-match-enrich-policy-type.html[geo_match enrichment policy]:
++
+[source,js]
+----------------------------------
+PUT /_enrich/policy/csa_lookup 
+{ 
+  "geo_match": { 
+    "indices": "csa", 
+    "match_field": "coordinates", 
+    "enrich_fields": [ "GEOID", "NAME"] 
+  } 
+}
+----------------------------------
+
+. To initialize the policy, run:
++
+[source,js]
+----------------------------------
+POST /_enrich/policy/csa_lookup/_execute
+----------------------------------
+
+. To create a ingest pipeline, run:
++
+[source,js]
+----------------------------------
+PUT _ingest/pipeline/lonlat-to-csa 
+{ 
+  "description": "Reverse geocode longitude-latitude to combined statistical area", 
+  "processors": [ 
+    { 
+      "enrich": { 
+        "field": "geo.coordinates", 
+        "policy_name": "csa_lookup", 
+        "target_field": "csa", 
+        "ignore_missing": true, 
+        "ignore_failure": true, 
+        "description": "Lookup the csa identifier" 
+      } 
+    }, 
+    { 
+      "remove": { 
+        "field": "csa.coordinates", 
+        "ignore_missing": true, 
+        "ignore_failure": true, 
+        "description": "Remove the shape field" 
+      } 
+    } 
+  ] 
+}
+----------------------------------
+
+. To update your existing data, run:
++
+[source,js]
+----------------------------------
+POST kibana_sample_data_logs/_update_by_query?pipeline=lonlat-to-csa
+----------------------------------
+
+. To run the pipeline on new documents at ingest, run:
++
+[source,js]
+----------------------------------
+PUT kibana_sample_data_logs/_settings 
+{ 
+  "index": { 
+    "default_pipeline": "lonlat-to-csa" 
+  } 
+}
+----------------------------------
+
+. Open the main menu, and click *Discover*.
+. Set the index pattern to *kibana_sample_data_logs*.
+. Open the <<set-time-filter, time filter>>, and set the time range to the last 30 days.
+. Scan through the list of *Available fields* until you find the `csa.GEOID` field. You can also search for the field by name.
+. Click image:images/reverse-geocoding-tutorial/add-icon.png[Add icon] to toggle the field into the document table.
+. Find the 'csa.NAME' field and add it to your document table.
+
+Your web log data now contains `csa.GEOID` and `csa.NAME` fields from the matching *csa* region. Web log traffic not contained in a CSA region does not have values for `csa.GEOID` and `csa.NAME` fields.
+
+[role="screenshot"]
+image::maps/images/reverse-geocoding-tutorial/discover_enriched_web_log.png[]
```

Comment:
This image is really small.  Is it necessary to show the list of Available fields?

Reply:
> Is it necessary to show the list of Available fields?

Not really. The relevant part is table but I think its still nice to show users where the fields came from and why they are in the table.